### ISSUE TITLE
Rule closure_end_indentation autocorrect false positive

### ISSUE THREAD

**Issue by Jeehut** [realm/SwiftLint#2354]
When autocorrecting this code via `swiftlint autocorrect`

```swift
        // MARK: Training
        static func trainingCreateDislike(exerciseId: Int) -> APIRequest<Void> {
            return SingleVoidAPIRequest(
                .trainingCreateDislike(exerciseId),
                customStatusCodeHandlers: [
                    412: { .failure(APIRequestError.createDislikeNotAllowed(statusCode: $0.statusCode)) }
                ]
            ).onSuccess { _ in
                APIRequestManager.shared.trainingDislikeCreatedEvent.raise(with: exerciseId)
            }
        }
```

SwiftLint auto-corrects the closing brace of the `onSuccess` closure by adding four spaces, which is wrong since the code was indented correctly. The result looks like this:

```swift
        // MARK: Training
        static func trainingCreateDislike(exerciseId: Int) -> APIRequest<Void> {
            return SingleVoidAPIRequest(
                .trainingCreateDislike(exerciseId),
                customStatusCodeHandlers: [
                    412: { .failure(APIRequestError.createDislikeNotAllowed(statusCode: $0.statusCode)) }
                ]
            ).onSuccess { _ in
                APIRequestManager.shared.trainingDislikeCreatedEvent.raise(with: exerciseId)
                } // << notice the additional spaces here
        }
```
